libnavigation-base/src/test/java/com/mapbox/navigation/base/internal/route/AnnotationsRefresherTest.kt

```diff
@@ -0,0 +1,304 @@
+                ),
+                arrayOf(
+                    defaultAnnotation,
+                    defaultAnnotation,
+                    0,
+                    defaultAnnotation,
+                    "2 default annotations, index = 0"
+                ),
+                arrayOf(
+                    defaultAnnotation,
+                    defaultAnnotation,
+                    5,
+                    defaultAnnotation,
+                    "2 default annotations, index = 5"
+                ),
+                arrayOf(
+                    annotationWithCongestionNumericOnlyEmpty,
+                    defaultAnnotation,
+                    0,
+                    defaultAnnotation,
+                    "Empty congestion_numeric + default, index = 0"
+                ),
+                arrayOf(
+                    annotationWithCongestionNumericOnlyEmpty,
+                    defaultAnnotation,
+                    0,
+                    defaultAnnotation,
+                    "Empty congestion_numeric + default, index = 5"
+                ),
+                arrayOf(
+                    defaultAnnotation,
+                    annotationWithCongestionNumericOnlyEmpty,
+                    0,
+                    annotationWithCongestionNumericOnlyEmpty,
+                    "Default + empty congestion_numeric, index = 0"
+                ),
+                arrayOf(
+                    defaultAnnotation,
+                    annotationWithCongestionNumericOnlyEmpty,
+                    5,
+                    LegAnnotation.builder().congestionNumeric(listOf(0, 0, 0, 0, 0)).build(),
+                    "Default + empty congestion_numeric, index = 5"
+                ),
+                arrayOf(
+                    annotationWithCongestionNumericOnlyEmpty,
+                    annotationWithCongestionNumericOnlyEmpty,
+                    0,
+                    annotationWithCongestionNumericOnlyEmpty,
+                    "Empty congestion_numeric x2, index = 0",
+                ),
+                arrayOf(
+                    annotationWithCongestionNumericOnlyFilled,
+                    defaultAnnotation,
+                    0,
+                    defaultAnnotation,
+                    "Filled congestion_numeric + default, index = 0"
+                ),
+                arrayOf(
+                    annotationWithCongestionNumericOnlyFilled,
+                    defaultAnnotation,
+                    5,
+                    defaultAnnotation,
+                    "Filled congestion_numeric + default, index = 5"
+                ),
+                arrayOf(
+                    defaultAnnotation,
+                    LegAnnotation.builder().congestionNumeric(listOf(3, 4, 5)).build(),
+                    2,
+                    LegAnnotation.builder().congestionNumeric(listOf(0, 0, 3, 4, 5)).build(),
+                    "Default + filled congestion_numeric, index = 2"
+                ),
```

Comment:
Is it how Directions API works or it's a test which verifies what happen if directions API returns incorrect result?

Reply:
It's a valid case. Directions API will only return new annotations. So if the index is 2 and the total number is 5, there will be 3 annotations in the answer.

Reply:
can leg's length change on the fly?

Reply:
Why? I don't know. What exactly do you mean by "on the fly"?

@mapbox/navigation-api is this correct?
> Directions API will only return new annotations. So if the index is 2 and the total number is 5, there will be 3 annotations in the answer.

Reply:
> What exactly do you mean by "on the fly"?

I meant that leg had 3 line segments before refresh and 5 after refresh, so it was changed while user were driving

Reply:
The number of annotations can change when specifying current index. I'll explain by example.
Lets suppose we have a route with geometry as `0 -- 1 -- 2 -- 3 -- 4 -- 5` where 0 & 5 are source & destination respectively. When you refresh with index=0, you get 5 annotations back. Now later, if you refresh with index=2, you might get more than 3 annotations (say 4). This happens due to snapping differences when the server tries to mapmatch from index 2 onward (sometimes it does not snap to the exact index). This is an issue with the server-side response and may be addressed in the future.

For now, when merging partial annotations, assume the first annotation corresponds to the current index supplied and any extra annotations beyond the end of current leg should be dropped.

Reply:
The number of annotations can change when specifying current index. I'll explain by example.
Lets suppose we have a route with geometry as `0 -- 1 -- 2 -- 3 -- 4 -- 5` where 0 & 5 are source & destination respectively. When you refresh with index=0, you get 5 annotations back. Now later, if you refresh with index=2, you might get more than 3 annotations (say 4). This happens due to snapping differences when the server tries to mapmatch from index 2 onward (sometimes it does not snap to the exact index). This is an issue with the server-side response and may be addressed in the future.

For now, when merging partial annotations, assume the first annotation corresponds to the current index supplied and any extra annotations beyond the end of current leg should be dropped. Eg for the case explained above, client should drop the last annotation if a total of 4 was returned (since only the first 3 will map to annotations in the original response).

Reply:
Thanks for the detailed explanation, I'll add a test to check this behaviour.